### PR TITLE
Fix compilation

### PR DESCRIPTION
**Pull request by DominicTobias-b1**
This is not being compiled properly resulting in the following issue: "Uncaught ReferenceError: exports is not defined" in upstream consumers https://stackoverflow.com/questions/42497479/uncaught-referenceerror-exports-is-not-defined-in-filed-generated-by-typescript

This is because all the files were emitting `Object.defineProperty(exports, "__esModule", { value: true });` which breaks if the upstream consumer is not wrapped in CommonJS crap which defines `exports`.

Also abstract interface files were being compiled like this:

```js
"use strict";
/**
 * @module RPC-API-Methods
 * copyright defined in eosjs/LICENSE.txt
 */
Object.defineProperty(exports, "__esModule", { value: true });
//# sourceMappingURL=eosjs-rpc-interfaces.js.map
```

Instead of like this:

```js
/**
 * @module RPC-API-Methods
 * copyright defined in eosjs/LICENSE.txt
 */
//# sourceMappingURL=eosjs-rpc-interfaces.js.map
```

I don't know why there are a bunch of builds with TS web & not web, and webpack (which isn't even running on NPM install) 🧐

### CODE REVIEW

**Comment by DominicTobias-b1** [comment]
I've tested this and fixes our problems in b1x-ui+cornea but I'm making it a major version so that any other upstreams know they should pay special attention when upgrading. The build setup of this project seems like a confusing mess to me.

File: src/eosjs-api.ts [hunk 1]
```diff
@@ -95,8 +95,8 @@ export class Api {
         this.textEncoder = args.textEncoder;
         this.textDecoder = args.textDecoder;
 
-        this.abiTypes = ser.getTypesFromAbi(ser.createInitialTypes(), abiAbi);
-        this.transactionTypes = ser.getTypesFromAbi(ser.createInitialTypes(), transactionAbi);
+        this.abiTypes = ser.getTypesFromAbi(ser.createInitialTypes(), abiAbi as any);
```
Comment (DominicTobias-b1) [inline src/eosjs-api.ts:98]: The types these functions expect don't match the json shapes 🤷‍♂️

**Review by sandeeprakhrablockone — APPROVED** [review]
:godmode: Fixing the official EOSJS library. That is GODMODE.

**Comment by DominicTobias-b1** [comment]
> :godmode: Fixing the official EOSJS library. That is GODMODE.
> In `ez-proxy` though we use this specific version which has some special Rodeos query functions exposed.
> 
> ```
> "eosjs": "^21.1.0-25-707787d"
> ```
> 
> I'm not sure where we actually use this fork you updated.

Interesting, we are using this fork in b1x-ui/cornea/ctrl. So I guess this means we have double the bundle size for this @ FYI @TaraTritt

**Pull request merged**